src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,178 @@
+        stmt.setInt(1, id);
+        stmt.execute();
+    }
+
+    public static ArrayList<Posts> selectAllMessages(Connection conn,String currentUser) throws SQLException {
+        ArrayList<Posts> messages = new ArrayList<>();
+        PreparedStatement stmt = conn.prepareStatement("SELECT * FROM posts INNER JOIN users ON posts.user_name = users.user_name WHERE users.user_name = ?");
+        stmt.setString(1,currentUser);
+        ResultSet results = stmt.executeQuery();
+        while (results.next()) {
+            int id = Integer.valueOf(results.getString("posts.postId"));
+            String name = results.getString("users.user_name");
+            String text = results.getString("posts.text");
+            Posts message = new Posts(id, name, text);
+            messages.add(message);
+        }
+        return messages;
+    }
+
+
+    public static void main(String[] args) throws SQLException {
+
+        Server.createWebServer().start();
+        Connection conn = DriverManager.getConnection("jdbc:h2:./main");
+        createTables(conn);
+
+        Spark.staticFileLocation("/templates");
+        Spark.init();
+
+        Spark.get("/",
+                ((request, response) -> {
+                    HashMap m = new HashMap();
+                    Session session = request.session();
+                    String name = session.attribute("userName");
+                    if (name == null) {
+                        return new ModelAndView(m, "login.html");
+                    }
+                    User user = selectUser(conn,name);
+                    ArrayList<Posts> msgs = selectAllMessages(conn,user.getUserName());
+                    m.put("posts", msgs);
+                    m.put("userName", user.getUserName());
+                    return new ModelAndView(m, "posts.html");
+
+                }),
+                new MustacheTemplateEngine()
+        );
+
+        Spark.post("/login",
+                ((request, response) -> {
+                    String loginName = request.queryParams("loginName");
+                    String password = request.queryParams("password");
+
+                    if(loginName==null||password==null){
+                        response.redirect("/");
+                        return"";
+                    }
+                    insertUser(conn,loginName,password);
+                    Session session = request.session();
+                    session.attribute("userName",loginName);
+                    response.redirect("/");
+                    return "";
+                })
+        );
+
+        Spark.post("/updateMessage",
+                (((request, response) -> {
+               int number = Integer.valueOf(request.queryParams("number")) ;
+              String post = request.queryParams("post");
+              updateMessage(conn,number,post);
+              response.redirect("/");
+              return "";
+                }))
+               );
```

Comment:
Almost Vegas.  Your update doesn't work as you have it now. I get  a 404 error which means the mapping is off. Also take a look at the SQL query in the delete method and compare that to update closely.  The messages show up oddly as well.  As for your question.  Fix the update method and it will meet the standard for Sessions and SQL Basics.  But you will need to write tests for your methods to meet the standard on SQL Joins.